db-tools-ij/src/main/java/com/splicemachine/db/impl/tools/ij/StatementFinder.java

```diff
@@ -31,549 +31,567 @@
+     * by this class as two strings, 'Mary' and 's lamb'.
+     * <p>
+     * The delimiter of the string is expected to be repeated at
+     * its other end. If the other flavor of delimiter occurs within
+     * the string, it is just a normal character within it.
+     * <p>
+     * All characters except the delimiter are permitted within the
+     * string. If EOF is hit before the closing delimiter is found,
+     * the end of the string is assumed. Parsers using this parser
+     * will detect the error in that case and return appropriate messages.
+     *
+     * @param stringDelimiter the starting and ending character
+     *                        for the string being read.
+     */
+    private void readString(char stringDelimiter) {
+        state = IN_STRING;
+        do {
+            char nextChar = readChar();
+
+            if (atEOF()) {
+                state = END_OF_INPUT;
+                return;
+            }
+
+            if (nextChar == stringDelimiter) {
+                // we've reached the end of the string
+                state = IN_STATEMENT;
+                return;
+            }
+
+            // still in string
+        } while (state == IN_STRING); // could be while true...
+    }
+
+    private boolean atEOF() {
+        return atEOF;
+    }
+
+    private boolean peekEOF() {
+        return peekEOF;
+    }
+
+    /**
+     * return the next character in the source stream and
+     * append it to the statement buffer.
+     *
+     * @return the next character in the source stream.
+     */
+    private char readChar() {
+        if (!peeked) peekChar();
+
+        peeked = false;
+        atEOF = peekEOF;
+
+        if (!atEOF) statement.append(peekChar);
+
+        return peekChar;
+    }
+
+    /**
+     * return the next character in the source stream, without
+     * advancing.
+     *
+     * @return the next character in the source stream.
+     */
+    private char peekChar() {
+        peeked = true;
+        char c = '\00';
+
+        try {
+            int cInt;
+
+            // REMIND: this is assuming a flat ascii source file.
+            // will need to beef it up at some future point to
+            // understand whether the stream is ascii or something else.
+            cInt = source.read();
+            peekEOF = (cInt == -1);
+            if (!peekEOF) c = (char) cInt;
+        } catch (IOException ie) {
+            throw ijException.iOException(ie);
+        }
+
+        peekChar = c;
+        return c;
+    }
+
+    public void setTerminator(char terminator) {
+        assert isValidTerminator(terminator);
+        this.terminator = terminator;
+    }
+
+    private static boolean isAsciiCharacter(char c) {
+        return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
+    }
+
+    public static boolean isValidTerminator(char terminator) {
+        return !isAsciiCharacter(terminator) && !DISALLOWED_TERMINATORS.contains(terminator);
```

Comment:
(minor comment) also ; is an ASCII character... i think you mean Character.isLetter(terminator)

Reply:
+1